### ISSUE TITLE
fe.protection.interact.*

### ISSUE THREAD

**Issue by kerts93** [ForgeEssentials/ForgeEssentialsMain#1270]
If you deny interact in zones, it works, until they left click items. Some mods like Jabba also have actions under left click (get stuff from barrel) but this doesn't protect from that.

**Comment by olee** [comment]
What does `/protectdebug` show if you left-click a barrel for example?

**Comment by kerts93** [comment]
Nothing actually.

**Comment by olee** [comment]
You need to run the command `/protectdebug` once and then interact with the barrel - it MUST show something.
Try doing some other stuff (e.g. placing a block) and you should definetly see some messages.

**Comment by kerts93** [comment]
Yes, doing other actions shows something, but left clicking does not. I can send you video if needed.

**Comment by olee** [comment]
Ok this probably means that Jabba is doing shit and added their event-listener with highest priority.
I'm gonna check it out if I have time and if it's true I'm gonna get that shit corrected :stuck_out_tongue_winking_eye:

**Comment by olee** [comment]
Uhh.... yeah - seems like Jabba source is private and I can't bother to find out if he / they have an issue tracker.
I could only try to force my way into the forge event bus and try to find the event that's triggered when interacting with the barrels.
If I find them I might be able to lower their priority with some injected code or whatever.

TL;DR: Blame Jabba

**Comment by kerts93** [comment]
Thank you :)
Here is also the video, to make sure I did not do something wrongly
http://youtu.be/zCu_Wmc-9vE

Also Jabba bucktracker:
https://bitbucket.org/ProfMobius/jabba/issues?status=new&status=open

**Comment by olee** [comment]
Thanks for the source link that helps a lot.
It seems the event is a simple block left-click event handler [source](https://bitbucket.org/ProfMobius/jabba/src/981e4a246e39faf462e88726355ebfe2b0b46ec0/src/main/java/mcp/mobius/betterbarrels/common/blocks/TileEntityBarrel.java?at=forge_1187#cl-197)

In general our protection should easily cover this. There must be some other stuff happening in Jabba that prevents the protection from working...

**Comment by taelnia** [comment]
It's standard MC Block interaction method overrides: [BlockBarrel.java](https://bitbucket.org/ProfMobius/jabba/src/981e4a246e39faf462e88726355ebfe2b0b46ec0/src/main/java/mcp/mobius/betterbarrels/common/blocks/BlockBarrel.java?at=forge_1187#cl-109)

Feel free to ask any questions about the source. It's certainly not private, and we're not doing anything crazy for block interaction.

**Comment by kerts93** [comment]
After some time we can can confirm that same problem applies with [Thermal Expansion Caches](http://teamcofh.com/thermal-expansion/items/cache-2/) . 

I turned on /protectdebug and nothing is shown when left click (actually I tried that left-click doesn't show anything for any of the items/blocks (It shows only when the item/block actually breaks), but there are quite a lot of mods that use only left-click for secondary interaction.

**Comment by olee** [comment]
I will try to investigate the problem once I have some time.

**Comment by olee** [comment]
Ok this was my fault.
I remember that these block-interactions worked once, but I added a bug when adding an additional check for interactions.
It works now without any problems.
I could even consider separating `fe.protection.interact` into  `fe.protection.interact.left` and  `fe.protection.interact.right` so both right- and left-click could be controlled individually.

**Comment by kerts93** [comment]
Thank you very much for fast and good work :)

**Comment by taelnia** [comment]
Glad to see it was resolved :smile:

For what it's worth, Your suggesting about separating permissions into both left and right click would probably be a good one to do.  A lot of mods these days are starting to add functionality on both interactions.

**Comment by olee** [comment]
Well the problem is we would have a lot more permissions to configure.
I need to talk about it with our development team and then we will see what we can do.

**Comment by taelnia** [comment]
For ease of configuring, perhaps leave existing combined option as is, but also add in the left and right options.  This would give three total options that would provide existing behaviour while allowing server operators the granularity of permissions they may need for some mods.

**Comment by olee** [comment]
Having the combined option would mean we need to do 2 permission checks for each interaction.
I want FE to have as few impact on server performance as possible so I'd rather have two permissions to configure than 2 permission checks.
Also having 3 different permission nodes would just confuse admins :smile:
